File: app/src/main/java/com/example/testhexagongame/utils/Iterator.java
package com.example.testhexagongame.utils;

import java.util.ArrayList;

public class Iterator<T> {
    private ArrayList<T> elements;
    private int index;

    public Iterator(ArrayList<T> elements) {
        this.elements = elements;
        this.index = 0;
    }

    public boolean hasNext() {
        return elements.size() > index;
    }
    public T getNext() {
        return elements.get(index++);
    }
    public void reset(){
        index = 0;
    }
}

File: app/src/main/java/com/example/testhexagongame/utils/Iterator.kt
package com.example.testhexagongame.utils

class Iterator<T>(private val elements: List<T>) {
    private var index = 0
    fun hasNext(): Boolean {
        return elements.size > index
    }

    fun getNext(): T {
        return elements[index++]
    }
    fun reset() {
        index = 0
    }
}